Created helpers for defining ConfigMaps, Pods, Containers and Volumes and Improved AMQP Clients

fgiorgetti opened the pull request.

nicob87 commented on pkg/framework/pod.go:
```diff
@@ -0,0 +1,144 @@
+	cb := new(ContainerBuilder)
+	cb.c = v1.Container{}
+	cb.c.Name = name
+	cb.c.Image = image
```
```
cb.c = v1.Container{Name: name, 
                                Image: image}
```
just an idea

fgiorgetti replied:
It had to be done this way because of the type composition nature of v1.Container.
Name and Image are inherited from other types, and cannot be set directly.

fgiorgetti commented:
This is ready for an initial review. I also have a test suite using the updated
code here just in case you guys need some more info.

nicob87 commented:
![image](https://user-images.githubusercontent.com/13121406/71526896-a88a8d00-28b7-11ea-87eb-eed2db9fd67a.png)

nicob87 commented:
a lot of work here!, I am using it and it works, just an "opinion", taking into account this: 
![image](https://user-images.githubusercontent.com/13121406/71527832-d8d42a80-28bb-11ea-86cd-e1150d4270bc.png)

maybe, the AmqpQEReceiver and AmqpQESender does not need to exists, and just using AmqpQEClientCommon.
You can have the AmqpQESenderBuilder and AmqpQEReceiverBuilder, assigning "MessageContent" and "MessageContentFromFile" attributes to the sender builder, since they are used only for building. and both, sender and receiver builders, could build just a AmqpQEClientCommon object instead of specific AmqpQEReceiver and QmqpQESender objects.

fgiorgetti commented:
@nicob87 You should send a PR adding these class diagrams to the repository. They are useful for discussing ideas.

nicob87 commented:
@fgiorgetti   (about the UML) take a look to !23

fgiorgetti commented:
> a lot of work here!, I am using it and it works, just an "opinion", taking into account this:
> ![image](https://user-images.githubusercontent.com/13121406/71527832-d8d42a80-28bb-11ea-86cd-e1150d4270bc.png)
> 
> maybe, the AmqpQEReceiver and AmqpQESender does not need to exists, and just using AmqpQEClientCommon.
> You can have the AmqpQESenderBuilder and AmqpQEReceiverBuilder, assigning "MessageContent" and "MessageContentFromFile" attributes to the sender builder, since they are used only for building. and both, sender and receiver builders, could build just a AmqpQEClientCommon object instead of specific AmqpQEReceiver and QmqpQESender objects.

I agree. I will modify it, so the AmqpQEClientCommon won't hold any property, instead the builder will populate the respective cmd line args directly.

fgiorgetti commented:
@nicob87 check it out again. I have removed both Sender and Receiver, kept just the common and removed properties from the amqp qe client common type.
Now specific stuff stays only on the builders, and they are passed to the sender or receiver instances
through command line only.
